examples/i18n/main.rs

```diff
@@ -0,0 +1,117 @@
+use std::collections::HashMap;
+use std::env;
+use std::fs::read_to_string;
+use clap::text_provider::TextProvider;
+use rust_i18n::set_locale;
+use rust_i18n::t;
+use rust_i18n::Backend;
+use std::path::PathBuf;
+use clap::{Parser, Subcommand};
+
+pub struct RemoteI18n {
+    pub trs: HashMap<String, HashMap<String, String>>,
+}
+
+impl RemoteI18n {
+    fn new() -> Self {
+        let en = read_to_string("./examples/i18n/locales/en.yml").unwrap();
+        let jp = read_to_string("./examples/i18n/locales/jp.yml").unwrap();
+        let en_trs = serde_yml::from_str::<HashMap<String, String>>(&en).unwrap();
+        let jp_trs = serde_yml::from_str::<HashMap<String, String>>(&jp).unwrap();
+        let trs = HashMap::from([("en".to_owned(), en_trs), ("jp".to_owned(), jp_trs)]);
+
+        Self {
+            trs
+        }
+    }
+}
+
+impl Backend for RemoteI18n {
+    fn available_locales(&self) -> Vec<&str> {
+        self.trs.keys().map(|k| k.as_str()).collect()
+    }
+
+    fn translate(&self, locale: &str, key: &str) -> Option<&str> {
+        self.trs.get(locale)?.get(key).map(|k| k.as_str())
+    }
+}
+
+rust_i18n::i18n!("locales", backend = RemoteI18n::new());
+
+
+#[derive(Parser)]
+#[command(version, about, long_about = None)]
+struct Cli {
+    /// {name-help-text}
```

Comment:
Notice that we need to provide a template string with the text lookup key inside in order to have Clap interpolate the text. This may not be ideal, but it was the simplest implementation as it didn't require making any changes to clap_derive.